filebeat/inputsource/common/listener.go

```diff
@@ -110,13 +114,17 @@ func (l *Listener) run() {
 		}
 
 		handler := l.handlerFactory(*l.config)
-		closer := WithCloser(l.closer, func() { conn.Close() })
+		ctx, cancel := context.WithCancel(l.ctx)
+		go func() {
+			<-ctx.Done()
+			conn.Close()
+		}()
```

Comment:
The handler go-routine is started always. Let's move cancellation setup local to the that go-routine. Consider to use `ctxtool.WithFunc`.

Reply:
Nice, I updated it to use `ctxtool.WithFunc`.